src/commands/Games/higherLower.ts

```diff
@@ -0,0 +1,267 @@
+				if (game.callback === null) return;
+
+				const emoji = this.resolveCollectedEmoji(message, game, reaction);
+				if (emoji === null) return;
+
+				game.callback(emoji);
+				game.callback = null;
+			}, async () => {
+				if (game.callback !== null) {
+					game.callback(null);
+					game.callback = null;
+				}
+				try {
+					await this.end(game, message, game.turn > 1);
+				} catch (error) {
+					this.client.emit(Events.Wtf, error);
+				}
+			}),
+			response,
+			running: true,
+			turn: 1,
+			number: this.random(0),
+			wager,
+			emojis: this.kFirstReactionArray,
+			callback: null
+		};
+
+		while (game.running) {
+			await game.response.reactions.removeAll();
+			// Send the embed
+			const { TITLE, DESCRIPTION, FOOTER } = message.language.tget('COMMAND_HIGHERLOWER_EMBED');
+			await game.response.edit(null, new MessageEmbed()
+				.setColor(getColor(message))
+				.setTitle(TITLE(game.turn))
+				.setDescription(DESCRIPTION(game.number))
+				.setFooter(FOOTER));
+
+			// Add the options
+			if (game.turn > 1) game.emojis = this.kReactionArray;
+			for (const emoji of game.emojis) {
+				await game.response.react(emoji);
+			}
+
+			const emoji = await this.awaitReactionAction(game);
+			if (emoji === null) break;
+
+			// Main game logic (at last)
+			const oldNum = game.number;
+			game.number = this.random(game.number);
+
+			switch (emoji) {
+				case HigherLowerReactions.Higher: {
+					game.running = await (game.number > oldNum ? this.win(game, message) : this.loss(game, message));
+					break;
+				}
+				case HigherLowerReactions.Lower: {
+					game.running = await (game.number < oldNum ? this.win(game, message) : this.loss(game, message));
+					break;
+				}
+				case HigherLowerReactions.Cashout: {
+					await this.end(game, message, true);
+					game.running = false;
+					break;
+				}
+			}
+
+			if (game.running) game.turn++;
+		}
+	}
+
+	private async win(game: HigherLowerGameData, message: KlasaMessage) {
+		const { language } = message;
+		await game.response.reactions.removeAll();
+
+		// TODO(Quantum): Implement Winning event for InfluxDB
+		const { TITLE, DESCRIPTION, FOOTER } = message.language.tget('COMMAND_HIGHERLOWER_WIN');
+		await game.response.edit(null, new MessageEmbed()
+			.setColor(getColor(message))
+			.setTitle(TITLE)
+			.setDescription(DESCRIPTION(this.calculateWinnings(game.wager, game.turn), game.number))
+			.setFooter(FOOTER));
+
+		// Ask the user whether they want to continue or cashout
+		game.emojis = this.kWinReactionArray;
+		for (const emoji of game.emojis) {
+			await game.response.react(emoji);
+		}
+
+		const reaction = await this.awaitReactionAction(game);
+		await game.response.reactions.removeAll();
+
+		const action = reaction === null
+			? EndingAction.Timeout
+			: reaction === HigherLowerReactions.Ok
+				? EndingAction.Play
+				: EndingAction.Stop;
+
+		// Decide whether we timeout, stop, or continue
+		switch (action) {
```

Comment:
No default cause is on purpose?

Reply:
Yeah, it's unreachable, but I can make it throw as 100% of the times it'll be a bug.